packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,503 @@
-    return docs.map(([doc, score]) => ({
-      document: doc as Document<Metadata>,
-      score,
-    }));
+  class StaticLoader<DocumentMetadata extends Jsonifiable = Jsonifiable> {
+    constructor(readonly documents: Document<DocumentMetadata>[], readonly pageSize?: number) {}
+
+    load(request: Request): Promise<Response<Document<DocumentMetadata>>> {
+      if (this.pageSize) {
+        const page: number = Number(request.pageToken ?? 0);
+        const start = page * this.pageSize;
+        const end = start + this.pageSize;
+        if (end > this.documents.length) {
+          return Promise.resolve(new Response(new Page(this.documents.slice(start))));
+        }
+        return Promise.resolve(new Response(new Page(this.documents.slice(start, end), (page + 1).toString())));
+      }
+
+      return Promise.resolve(new Response(new Page(this.documents)));
+    }
+  }
+
+  export function staticLoader<DocumentMetadata extends Jsonifiable = Jsonifiable>(
+    documents: Document<DocumentMetadata>[],
+    pageSize?: number
+  ): Loader<DocumentMetadata> {
+    const loader = new StaticLoader(documents, pageSize);
+    return (request) => loader.load(request);
   }
 }
 
+/** A function that splits a document into multiple chunks, for example to ensure it fits in appropriate context windows. */
+export type Chunker<
+  DocumentMetadata extends Jsonifiable = Jsonifiable,
+  ChunkMetadata extends Jsonifiable = Jsonifiable
+> = (document: Document<DocumentMetadata>) => Promise<Chunk<ChunkMetadata>[]>;
+
 /**
- * If you need to chunk your documents, start with this chunker. Move to more powerful chunkers only as needed.
+ * A simple token size based text chunker. This is a good starting point for text documents.
  */
-export const defaultChunker = <Metadata extends Jsonifiable = Jsonifiable>(
-  doc: Document<Metadata>,
-  opts: ConstructorParameters<typeof TokenTextSplitter>
-) => {
+export const defaultChunker = async <Metadata extends Jsonifiable = Jsonifiable>(doc: Document<Metadata>) => {
   const splitter = new TokenTextSplitter({
     encodingName: 'gpt2',
     chunkSize: 600,
     chunkOverlap: 100,
-    ...opts,
   });
 
-  // This is complaining because {} doesn't necessary satisfy Metadata, since Metadata could have required fields.
-  // However, I think it's unlikely to cause a problem in practice at this point.
-  // @ts-expect-error
-  doc.metadata = doc.metadata ?? {};
+  const lcDocs = await splitter.createDocuments(doc.pageContent);
+  const chunks = lcDocs.map(
+    (lcDoc) =>
+      ({
+        content: lcDoc.pageContent,
+        documentName: doc.name,
+        metadata: doc.metadata,
+      } as Chunk<Metadata>)
+  );
 
-  return splitter.splitDocuments([
-    // TS doesn't realize that we've ensured that the metadata is defined.
-    // @ts-expect-error
-    doc,
-  ]);
+  return chunks;
 };
 
+/** A chunk is a piece of a document that's appropriately sized for an LLM's context window and for semantic search. */
```

Comment:
For `context window`, you can link to https://docs.ai-jsx.com/guides/brand-new#context-window.